.gitignore

```
@@ -51,8 +51,40 @@ wheels/
 # Checksums
 *.sha256
 
-# Data files (add specific patterns as needed)
-# *.csv
-# *.xlsx
-# *.parquet
-# data/
+# Data files - common formats
+*.csv
+*.xlsx
+*.xls
+*.parquet
+*.feather
+*.hdf5
+*.h5
+*.pkl
+*.pickle
+*.db
+*.sqlite
+*.sqlite3
+*.dta
+*.sav
+*.rds
+*.rda
+*.RData
+
+# Data directories (except .gitkeep files and READMEs)
+data/*
+!data/.gitkeep
+!data/README.md
+!data/*/
+data/*/*
+!data/*/.gitkeep
```

I think this could be shortened to:
```suggestion
data/**/* # Ignore all files in data directory recursively
!data/**/*.gitkeep # But keep all .gitkeep files
!data/**/README.md # And keep all README.md files
```
which also safeguards against data being put into directories at any depth